java/Nikiforov-Oleksandr/src/com/zoolatech/lecture4/tasks/_2/BinarySearchTree.java

```diff
@@ -1,43 +1,43 @@
 package com.zoolatech.lecture4.tasks._2;
 
 public class BinarySearchTree {
-    static class Node {
+    private class Node {
         int key;
         Node left, right;
 
         public Node(int key) {
             this.key = key;
-            left = right = null;
         }
     }
 
     private Node root;
 
     public BinarySearchTree(int value) {
-        root = new Node(value);
+        new Node(value);
```

Comment:
I was talking about the method `insertRec`, in the constructor you need to keep:

```
root = new Node(value);
```